src/llv8.h

```diff
@@ -7,6 +7,14 @@
 
 #include "src/llv8-constants.h"
 
+using lldb::ByteOrder;
+
+#if __BYTE_ORDER == __LITTLE_ENDIAN
+#define HOST_BYTE_ORDER ByteOrder::eByteOrderLittle
+#else
+#define HOST_BYTE_ORDER ByteOrder::eByteOrderBig
+#endif
```

Comment:
I'm changing to the little endian define to use `__ORDER_LITTLE_ENDIAN__` since `__BYTE_ORDER__` and `__ORDER_LITTLE_ENDIAN__` are both built in to gcc and clang. You can compile the program below as is (with no extra includes) so these should be available by default.
```
int main(int argc, char **argv)
{
    return __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__;
}
```
I don't think there's any point working it out at run time if you can work it out at compile time. (The compiler might be clever and work it out anyway.